### PR TITLE
tests: print rockcraft log files on spread failure

### PR DESCRIPTION
**Pull request by mr-cal**
- [X] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

Print rockcraft log files when spread fails, to capture more details in scenarios like [this](https://github.com/canonical/rockcraft/actions/runs/3143892833/jobs/5109299391)

### CODE REVIEW

File: spread.yaml [hunk 1]
```diff
@@ -73,6 +73,17 @@ restore-each: |
   lxc --project=rockcraft stop --all
   lxc --project=rockcraft delete $(lxc --project=rockcraft list -c n --format csv)
 
+debug-each: |
+  # output latest rockcraft log file on test failure
+  rockcraft_log_file=$(find /root/.cache/rockcraft/log/ -name 'rockcraft*.log' | sort -n | tail -n1)
+  if [[ -f $rockcraft_log_file ]]; then
+    echo "rockcraft log file contents ($rockcraft_log_file)"
```
Comment (tigarmo) [inline spread.yaml]:
```suggestion
    echo ">>>>>> rockcraft log file contents ($rockcraft_log_file)"
```
(just to make it easier to find this in the sea of output)

**Review by tigarmo — APPROVED** [review]
that was a good idea, thanks!

File: spread.yaml (hunk 1 above)
Comment (mr-cal) [reply spread.yaml]: Good idea, let me try some line breaks

**Comment by mr-cal** [comment]
[output](https://github.com/canonical/rockcraft/actions/runs/3145158849/jobs/5112208544):
```
rockcraft log file contents:
----------------------------
+ cat /root/.cache/rockcraft/log/rockcraft-20220928-163605.679561.log
2022-09-28 16:36:05.696 Starting Rockcraft 0.0.1.dev1
2022-09-28 16:36:05.720 Bad rockcraft.yaml content:
- plugin not registered: 'let-me-fail!' in field 'parts.issue-44-dir-owner'
2022-09-28 16:36:05.721 Traceback (most recent call last):
2022-09-28 16:36:05.721   File "/snap/rockcraft/x1/lib/python3.8/site-packages/rockcraft/project.py", line 330, in unmarshal
2022-09-28 16:36:05.721     project = Project(**data)
2022-09-28 16:36:05.721   File "pydantic/main.py", line 331, in pydantic.main.BaseModel.__init__
2022-09-28 16:36:05.721 pydantic.error_wrappers.ValidationError: 1 validation error for Project
2022-09-28 16:36:05.721 parts -> issue-44-dir-owner
2022-09-28 16:36:05.721   plugin not registered: 'let-me-fail!' (type=value_error)
2022-09-28 16:36:05.721 Full execution log: '/root/.cache/rockcraft/log/rockcraft-20220928-163605.679561.log'
+ echo ----------------------------
----------------------------
```

**Pull request merged**